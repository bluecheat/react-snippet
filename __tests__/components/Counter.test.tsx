import { render } from '@testing-library/react';
import Counter from 'pages/example/counter';


describe('<Counter / > test', () => {
	it('matches snapshopt', () => {
		const component = render(<Counter / >);
		expect(component.container).toMatchSnapshot(); // snapshot match
	});

	it('counter increments and decrements when the buttons are clicked', () => {
		const {container} = render(<Counter />)
		const [increment, decrement] = container.querySelectorAll('button')
		const message = container.firstChild.querySelector('h2')

		expect(message.textContent).toBe('0')
		fireEvent.click(increment)
		expect(message.textContent).toBe('1')
		fireEvent.click(decrement)
		expect(message.textContent).toBe('0')
	})
});