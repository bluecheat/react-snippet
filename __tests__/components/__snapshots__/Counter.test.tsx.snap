// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter / > test matches snapshopt 1`] = `
<div>
  <div>
    <h1>
      Counter
    </h1>
    <h2>
      0
    </h2>
    <button>
      +
    </button>
    <button>
      -
    </button>
  </div>
</div>
`;
